core/common/src/thrift/exception.thrift

```diff
@@ -1,5 +1,120 @@
 namespace java alluxio.thrift
 
+enum TExceptionStatus {
+  // Canceled indicates the operation was cancelled (typically by the caller).
+  CANCELED = 1,
```

Comment:
one high level comment: can we define these in proto so that the exception story can be shared between Thrift servers and data servers?